cordapp-cpk/src/main/kotlin/net/corda/plugins/cpk/CordappPlugin.kt

```diff
@@ -125,10 +125,6 @@ class CordappPlugin @Inject constructor(private val layouts: ProjectLayout): Plu
                         // to our own compile classpath.
                         // WE ARE MUTATING THESE DEPENDENCIES FOR EVERY CONFIGURATION THEY APPEAR IN!
                         dep.isTransitive = false
-
-                        // We also need to GUARANTEE that Gradle uses the jar artifact here.
-                        // Only the jar contains the OSGi metadata we need.
-                        dep.attributes(attributor::forJar)
```

Comment:
if I don't do this I get
```
Caused by: org.gradle.api.InvalidUserCodeException: Cannot set artifact / configuration information on a dependency that has attributes or capabilities configured
```
:-(

Reply:
Unfortunately we need that line.

Reply:
I think I now understand what is happening here. We only need to add the variant attribute to a `ProjectDependency` because nothing downloaded from a Maven repository will have a `classes` directory. However, it also appears that we _must not_ add a variant attribute to external dependencies because this would force everyone to publish Gradle metadata into those Maven repositories too :scream:.

I have added some extra tweaks to #446 so that the `cordaCPK` configuration is also "variant aware". This should allow us to handle CPK dependencies more nicely in the `cordapp-cpb` plugin. However, I have also discovered that any `ProjectDependency` created in the `withDependencies()` handler isn't automatically linked to the `cpk` task that builds it! This is a problem that I am frantically investigating.